### PR TITLE
Cleanup parquet tests

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

Cleans up some test duplication noticed whilst reviewing #3103

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: parquet/src/file/writer.rs [hunk 1]
```diff
@@ -1212,16 +1231,19 @@ mod tests {
...
-            let res = iter
-                .map(|elem| elem.get_int(0).unwrap())
-                .collect::<Vec<i32>>();
+            let res: Vec<_> = iter.map(&value).collect();
             assert_eq!(res, *item);
         }
         file_metadata
     }
 
-    fn assert_send<T: Send>(t: T) -> T {
```
Comment (alamb) [inline parquet/src/file/writer.rs:1223]: is this check covered elsewhere ?

File: parquet/src/file/writer.rs [hunk 2]
```diff
@@ -1245,58 +1267,17 @@ mod tests {
...
+    }
 
-        assert_eq!(reader.num_row_groups(), data.len());
-        assert_eq!(
-            reader.metadata().file_metadata().num_rows(),
-            rows,
-            "row count in metadata not equal to number of rows written"
+    #[test]
+    fn test_boolean_roundtrip() {
+        let my_bool_values: Vec<_> = (0..2049).map(|idx| idx % 2 == 0).collect();
```
Comment (alamb) [inline parquet/src/file/writer.rs:1275]: I double checked that `2049` is the same size as `parquet/tests/boolean_writer.rs`

File: parquet/tests/boolean_writer.rs [hunk 3]
```diff
@@ -1,89 +0,0 @@
...
-        let rg_md = row_group_writer.close().expect("close row group");
-        println!("total rows written: {}", rg_md.num_rows());
-    }
-    writer.close().expect("close writer");
-
-    let bytes = fs::read(path).expect("read file");
-    assert_eq!(&bytes[0..4], &[b'P', b'A', b'R', b'1']);
-
-    // Now that we have written our data and are happy with it, make
-    // sure we can read it back in < 5 seconds...
```
Comment (alamb) [inline parquet/tests/boolean_writer.rs:71]: 🤯  not sure what this was all about

**Review by alamb — APPROVED** [review]
LGTM -- my only hesitation is the removal of `assert_send` but otherwise nice cleanup

File: parquet/src/file/writer.rs (hunk 1 above)
Comment (tustvold) [reply parquet/src/file/writer.rs:1223]: Yes, all the async plumbing requires it

**Pull request merged**